### PR TITLE
Cycle sort in cpp

### LINKED ISSUE #587: Cycle Sort
I can implement Cycle sort algorithm in C++ as part of  GSSOC'21. Kindly assign this issue to me. @tarun26091999

### PR DESCRIPTION
**Pull request by Harikrishnan6336**
Fixes #587 Kindly review this @raksha009 @rudrakshi99 @plazzy99 @tarun26091999

### CODE REVIEW

File: Code/C++/cycle_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
+// C++ program to implement Cycle Sort algorithm
+/* Cycle Sort is an unstable in-place sorting algorithm. It is based on the 
+idea that one of the permutation of the numbers is in sorted order */
```
Comment (OjusWiZard) [inline Code/C++/cycle_sort.cpp:3]: Also, write the intuition behind the working of this algorithm.

File: Code/C++/cycle_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    cycle_sort(arr, n);
+    cout << "The sorted elements are : ";
+    for (int i = 0; i < n; i++)
+        cout << arr[i] << " ";
+    return 0;
+}
+
+/*
+Time Complexity - O(n^2), where 'n' is the size of the array
+Space Complexity - O(n)
```
Comment (OjusWiZard) [inline Code/C++/cycle_sort.cpp:73]: Space complexity seems incorrect to me

File: Code/C++/cycle_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    int start, ele, pos, temp;
+
+    // Traverse the entire array
+    for (start = 0; start <= n - 2; start++)
+    {
+        ele = a[start];
+        pos = start;
+        for (int i = start + 1; i < n; i++)
+            if (a[i] < ele)
+                pos++;
```
Comment (OjusWiZard) [inline Code/C++/cycle_sort.cpp:19]: Write the purpose of this loop as a comment.

File: Code/C++/cycle_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        // If element is present in right position
+        if (pos == start)
+            continue;
+
+        while (ele == a[pos])
+            pos += 1;
```
Comment (OjusWiZard) [inline Code/C++/cycle_sort.cpp:26]: Write the purpose of this loop as a comment.

File: Code/C++/cycle_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+Time Complexity - O(n^2), where 'n' is the size of the array
+Space Complexity - O(n)
+
+SAMPLE INPUT AND OUTPUT
+
+SAMPLE I
+
+How many numbers do you want to sort? 5
+Enter the numbers : 4 2 5 1 3
+The sorted elements are :  1 2 3 4 5
```
Comment (OjusWiZard) [inline Code/C++/cycle_sort.cpp:81]: Write at least **two** test cases.

**Review by OjusWiZard — CHANGES_REQUESTED** [review]

**Comment by OjusWiZard** [comment]
Resolve the conflict.

**Review by OjusWiZard — APPROVED** [review]

**Comment by OjusWiZard** [comment]
Resolve the conflict again pls

**Comment by Harikrishnan6336** [comment]
@OjusWiZard Done

**Pull request merged**